debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -0,0 +1,202 @@
+        logMinerProcessor = new LogMinerProcessor(schema, dmlParser);
+    }
+
+    /**
+     * This is the loop to get changes from LogMiner
+     *
+     * @param context change event source context
+     * @throws InterruptedException an exception
+     */
+    @Override
+    public void execute(ChangeEventSourceContext context) throws InterruptedException {
+
+        ResultSet res = null;
+
+        try (Connection connection = jdbcConnection.connection();
+             PreparedStatement fetchChangesFromMiningView =
+                     connection.prepareStatement(SqlUtils.queryLogMinerContents(connectorConfig.getSchemaName()))) {
+            long lastProcessedScn = offsetContext.getScn(); // last processes Scn
+            long oldestScnInOnlineRedo = LogMinerHelper.getFirstOnlineLogScn(connection);
+            if (lastProcessedScn < oldestScnInOnlineRedo) {
+                throw new RuntimeException("Online REDO LOG files don't contain the offset SCN. Clean offset info and start over");
+            }
+
+            LogMinerHelper.setNlsSessionParameters(jdbcConnection);
+
+            // 1. Configure Log Miner to mine online logs
+            LogMinerHelper.buildDataDictionary(connection);// todo should we renew it after a DDL?
+            LogMinerHelper.addOnlineRedoLogFilesForMining(connection);
+
+            // 2. Querying LogMinerRowLcr(s) from Log miner while running
+            while (context.isRunning()) {
+                LOGGER.trace("Receiving a change from LogMiner");
+
+                lastProcessedScn = offsetContext.getScn();
+                long currentScn = LogMinerHelper.getCurrentScn(connection);
+                LOGGER.debug("\nstart, lastProcessedScn: " + lastProcessedScn + ", endScn: " + currentScn);
+                if (lastProcessedScn == currentScn) {
+                    LOGGER.debug("No changes to process...pausing for a second");
+                    Metronome metronome = Metronome.sleeper(Duration.ofSeconds(1L), clock);
+                    metronome.pause();
+                }
+
+                LogMinerHelper.startOnlineMining(connection, lastProcessedScn, currentScn);
+
+                fetchChangesFromMiningView.setLong(1, lastProcessedScn);
+                fetchChangesFromMiningView.setLong(2, lastProcessedScn);
+
+                debugInfo(connection, "before");
+                res = fetchChangesFromMiningView.executeQuery();
+                debugInfo(connection, "after");
+
+                dispatchChanges(res);
+                offsetContext.setScn(currentScn);
+
+                res.close();
+            }
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        } finally {
+            // 3. disconnect
+            try (Connection connection = jdbcConnection.connection()) {
+                LogMinerHelper.endMining(connection);
+            } catch (SQLException e) {
+                LOGGER.error("Cannot close Log Miner session gracefully: " + e.getMessage());
```

Comment:
Please keep the original exception:
```suggestion
                LOGGER.error("Cannot close Log Miner session gracefully", e);
```